src/zenml/zen_stores/schemas/pipeline_run_schemas.py

```diff
@@ -283,3 +290,43 @@ def update(
 
         self.updated = datetime.utcnow()
         return self
+
+    def update_placeholder(
+        self, request: "PipelineRunRequestModel"
+    ) -> "PipelineRunSchema":
+        """Update a placeholder run.
+
+        Args:
+            request: The pipline run request which should replace the
+                placeholder.
+
+        Raises:
+            RuntimeError: If the DB entry does not represent a placeholder run.
+            ValueError: If the run request does not match the deployment or
+                pipeline ID of the placeholder run.
+
+        Returns:
+            The updated `PipelineRunSchema`.
+        """
+        if self.orchestrator_run_id:
```

Comment:
Might also make sense to check for `self.status` here instead and set `self.status = ExecutionStatus.RUNNING` if the placeholder was updated successfully

Reply:
Yes definitely, I added a new `initializing` execution status now and set it to the status of the replacement request here.